test/parse.test.ts

```diff
@@ -687,6 +710,86 @@ See more help with --help`)
     })
   })
 
+  describe('exactlyOne', () => {
+    it('throws if neither is set', () => {
+      expect(() => {
+        parse([], {
+          flags: {
+            foo: flags.string({exactlyOne: ['bar']}),
+            bar: flags.string({char: 'b', exactlyOne: ['foo']}),
+          },
+        })
+      }).to.throw()
+    })
+
+    it('throws if multiple are set', () => {
+      expect(() => {
+        parse(['--foo', 'a', '--bar', 'b'], {
+          flags: {
+            foo: flags.string({exactlyOne: ['bar']}),
+            bar: flags.string({char: 'b', exactlyOne: ['foo']}),
+          },
+        })
+      }).to.throw('--bar= cannot also be provided when using --foo=')
+    })
+
+    it('succeeds if exactly one', () => {
+      const out = parse(['--foo', 'a', '--else', '4'], {
+        flags: {
+          foo: flags.string({exactlyOne: ['bar']}),
+          bar: flags.string({char: 'b', exactlyOne: ['foo']}),
+          else: flags.string({char: 'e'}),
+        },
+      })
+      expect(out.flags.foo).to.equal('a')
+    })
+
+    it('succeeds if exactly one (the other option)', () => {
+      const out = parse(['--bar', 'b', '--else', '4'], {
+        flags: {
+          foo: flags.string({exactlyOne: ['bar']}),
+          bar: flags.string({char: 'b', exactlyOne: ['foo']}),
+          else: flags.string({char: 'e'}),
+        },
+      })
+      expect(out.flags.bar).to.equal('b')
+    })
+
+    it('succeeds if exactly one of three', () => {
```

Comment:
What if you had something like:
```
flags: {
          foo: flags.string({exactlyOne: ['bar']}),
          bar: flags.string({char: 'b', exactlyOne: ['else']}),
          else: flags.string({char: 'e'}),
        },
```
Granted I'm not sure why someone would do that, but I think your filter in parse.ts:90 will allow that, which I guess is technically right? So `--foo --bar` would fail, `--bar --else` would fail but `--foo --else` would pass? Might be worth to have a test for that if we think it is a valid scenario?

Reply:
definitely could make a test.
Struggling to imagine the goal of that usage, but that doesn't mean anything.

Reply:
Ya, I'm struggling to envision the use case too, but I guess if the code allows it someone will do it, so its probably good to have a test to make sure we don't break it in the future, OR don't allow it in the first place.

Reply:
will do.

Reply:
added test for that